jena-arq/src/main/java/org/apache/jena/sparql/exec/QueryExec.java

```diff
@@ -54,6 +54,12 @@ public static QueryExecBuilder graph(Graph graph) {
         return QueryExecDatasetBuilder.create().graph(graph);
     }
 
+    /** Create a {@link QueryExecBuilder} initialized with an empty dataset. */
+    public static QueryExecBuilder emptyDataset() {
```

Comment:
This is only used from `TestQueryExecution`. Please can the code go there so it is not in the runtime API.? 

It can use`DatasetGraphFactory.empty()`.

```java
return QueryExec.dataset(DatasetGraphFactory.empty()`);
```

Reply:
It was just a suggestion for a shortcut - I'll remove it.